raster/r.smooth.edgepreserve/r.smooth.edgepreserve.md

```diff
@@ -0,0 +1,119 @@
+  automatically clamped between `0` and `0.25` for numeric stability.
+  - For **Tukey**, the full range is used.
+
+- **`steps`**  
+  Specifies the number of smoothing iterations. Higher values result
+  in stronger smoothing.  
+  Note that:
+  - Excessive iterations can cause loss of fine detail.
+  - Detail preservation improves in this order:  
+    `quadratic` → `exponential` → `Tukey` → `Tukey` with **`-p`** flag.
+
+## NOTES
+
+Internally calculations are performed in double precision and only at the end
+converted to the same type as the input map.
+
+The module will try to keep all temporary data in RAM. Thus it is important
+to set the **`memory`** parameter as high as possible.  If the map does
+not fit into RAM, a temporary file will be used instead.
+
+The original *Perona & Malik* paper uses von Neumann (4-connected)
+neighbourhood for value calculation, but this module uses Moore
+(8-connected) neighbourhood. Computed gradients of neighbouring cells
+are adjusted to equalise distances for diagonals and non-square cells.
+
+## EXAMPLES
+
+```sh
+# Set computational region to orthophoto map
+g.region raster=ortho_2001_t792_1m -p
+
+# Smooth with average in a 3x3 window. Note how all edges have became blurry
+# but at the same time streets and roofs are still noisy
+r.neighbors input=ortho_2001_t792_1m output=ortho_smoothed_avg\
+ size=3 method=average
+
+# Smooth with median in a 3x3 window. Although better than average smoothing,
+# thin lines still are lost and noise on streets and roofs is still present.
+r.neighbors input=ortho_2001_t792_1m output=ortho_smoothed_med\
+ size=3 method=median
+
+# Smooth with quadratic diffusivity function. Note better preservation of
+# small details and reduction of noise on streets and roofs.
+r.smooth.edgepreserve function=quadratic input=ortho_2001_t792_1m \
+ output=ortho_smoothed_qa threshold=15 lambda=0.4 steps=20
+
+# Smooth with exponential diffusivity function. Even better edge delineation
+# but at the same time increase of noise in really noisy areas.
+r.smooth.edgepreserve function=exponential input=ortho_2001_t792_1m \
+ output=ortho_smoothed_ex threshold=15 lambda=0.4 steps=20
+
+# Smooth with aggressive Tukey's diffusivity function. Better preservation of
+# minor details e.g. as road markings but little smoothing in areas with
+# fine, well expressed texture.
+r.smooth.edgepreserve function=tukey input=ortho_2001_t792_1m \
+ output=ortho_smoothed_ta threshold=15 lambda=0.4 steps=20
+
+# Smooth with preserving Tukey's diffusivity function. Only low noise areas
+# have been smoothed.
+r.smooth.edgepreserve function=tukey input=ortho_2001_t792_1m \
+ output=ortho_smoothed_tp threshold=15 lambda=0.4 steps=20 -p
+```
+
+## SEE ALSO
+
+- Smooth with statistics: *[r.neighbours](r.neighbours)*
+- The Mumford-Shah variational model for image segmentation (an add-on):
+*[r.smooth.seg](https://grass.osgeo.org/grass84/manuals/addons/r.smooth.seg.html)*
```

Comment:
This should be at least grass-stable to not get obsolete.

Reply:
Yes, now we have unversioned links too. Fixed this.